### PR TITLE
Add transition example

### PR DESCRIPTION
**Pull request by wbamberg**
Here's an attempt at an example for `transition`.

It's pretty much an interactivisation of the existing example in the page: https://developer.mozilla.org/en-US/docs/Web/CSS/transition

### CODE REVIEW

**Comment by wbamberg** [comment]
Actually, wait, this isn't ready. Sorry.

**Comment by wbamberg** [comment]
OK, it's ready now.

I've thought before with these CSS examples actually - it would be useful sometimes for the example to be able to supply a JS function that's called when a new choice is selected. In this case it could change the transitioning properties, so the user doesn't need to hover over the output - they would just see the transition immediately on selection. Maybe there's some way to do this anyway...

File: live-examples/css-examples/transitions/transition.css [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#example-element {
+    background-color: #E4F0F5;
```
Comment (schalkneethling) [inline live-examples/css-examples/transitions/transition.css]: Super Nit: lowercase e and f

File: live-examples/css-examples/transitions/transition.css [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    padding: 1rem;
+    border-radius: .5rem;
+    font: 1em monospace;
+    width: 100%;
+    transition: margin-right 2s;
+}
+
+#default-example:hover>#example-element {
+    background-color: #990099;
+    color: #FFF;
```
Comment (schalkneethling) [inline live-examples/css-examples/transitions/transition.css]: Nits: colour can be `#909` and lowercase f

File: live-examples/css-examples/transitions/transition.html [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+<div class="example-choice">
+<pre><code id="example_three" class="language-css">transition: margin-right 2s ease-in-out;</code></pre>
+<button type="button" class="copy hidden" aria-hidden="true">
+    <span class="visually-hidden">Copy to Clipboard</span>
+</button>
+</div>
+
+<div class="example-choice">
+<pre><code id="example_four" class="language-css">transition: margin-right 2s ease-in-out 0.5s;</code></pre>
```
Comment (schalkneethling) [inline live-examples/css-examples/transitions/transition.html]: I believe our code style is to write fractions as `.5s`, excluding the 0

File: live-examples/css-examples/transitions/transition-timing-function.html [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+<div class="example-choice">
+<pre><code id="example_three" class="language-css">transition-timing-function: steps(6, end);</code></pre>
+<button type="button" class="copy hidden" aria-hidden="true">
+    <span class="visually-hidden">Copy to Clipboard</span>
+</button>
+</div>
+
+<div class="example-choice">
+<pre><code id="example_four" class="language-css">transition-timing-function: cubic-bezier(.29,1.01,1,-0.68);</code></pre>
```
Comment (schalkneethling) [inline live-examples/css-examples/transitions/transition-timing-function.html]: Nit, code style: spaces after commas.

**Comment by schalkneethling** [comment]
> In this case it could change the transitioning properties, so the user doesn't need to hover over the output - they would just see the transition immediately on selection. Maybe there's some way to do this anyway...

I believe you can already to this by specifying a `jsExampleSrc` in `meta.json` that will then be stuck into the template here[1]

[1] https://github.com/mdn/interactive-examples/blob/master/tmpl/live-css-tmpl.html#L25

**Comment by wbamberg** [comment]
Thanks @schalkneethling , I hope I got them all.

**Pull request merged**